lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -157,6 +158,10 @@ def add_resource_relationships(attrs, serializer, options = {})
             end
           end
         end
+
+        def attributes_for_top_level_links(serializer)
```

Comment:
Idk about using `attributes` here, maybe other method name? what you think about it? it's not actual attributes.